gensim/models/tfidfmodel.py

```diff
@@ -176,36 +206,33 @@ def smartirs_wglobal(docfreq, totaldocs, global_scheme):
         Calculated global weight.
 
     """
-
     if global_scheme == "n":
-        return 1.
-    elif global_scheme == "t":
+        return 1.0
+    elif global_scheme == "f":
         return np.log2(1.0 * totaldocs / docfreq)
+    elif global_scheme == "t":
+        return np.log2((totaldocs + 1.0) / docfreq)
     elif global_scheme == "p":
         return max(0, np.log2((1.0 * totaldocs - docfreq) / docfreq))
 
 
+@deprecated("Function will be removed in 4.0.0")
 def smartirs_normalize(x, norm_scheme, return_norm=False):
     """Normalize a vector using the normalization scheme specified in `norm_scheme`.
-
     Parameters
     ----------
     x : numpy.ndarray
-        Input array
+        The tf-idf vector.
     norm_scheme : {'n', 'c'}
-        Normalizing function to use:
-        `n`: no normalization
-        `c`: unit L2 norm (scale `x` to unit euclidean length)
+        Document length normalization scheme.
     return_norm : bool, optional
         Return the length of `x` as well?
-
```

Comment:
Does this render correctly? I remember missing blank lines in between documentation sections can mess up the rending (as well as make the code harder to read).

I'd prefer to retain the section separator blanks, for readability reasons.

Reply:
This is definitely an error, which must have happened during my editing. Fixed in 78746020.